package proto_ff;

import "yd_fieldoptions.proto";


message E_RankingRanking
{
	optional int32 m_rankingid = 1[(yd_fieldoptions.field_cname) = "排行榜ID"];
	optional string m_rankingname = 2[(yd_fieldoptions.field_cname) = "排行名称", (yd_fieldoptions.field_bufsize) = 64];
	optional string m_title = 3[(yd_fieldoptions.field_cname) = "榜一称号", (yd_fieldoptions.field_bufsize) = 64];
	optional int32 m_capacity = 4[(yd_fieldoptions.field_cname) = "实际排多少元素"];
	optional int32 m_showsize = 5[(yd_fieldoptions.field_cname) = "显示多少个元素"];
	optional int32 m_rankhide = 6[(yd_fieldoptions.field_cname) = "是否在排行榜隐藏"];
	repeated int32 m_dateid = 7[(yd_fieldoptions.field_cname) = "排行数据", (yd_fieldoptions.field_arysize) = 5];
}


message Sheet_RankingRanking
{
	repeated E_RankingRanking E_RankingRanking_List = 1[(yd_fieldoptions.field_arysize)=16];
}

message E_RankingDate
{
	optional int32 m_dateid = 1[(yd_fieldoptions.field_cname) = "数据ID"];
	optional string m_datename = 2[(yd_fieldoptions.field_cname) = "数据名称", (yd_fieldoptions.field_bufsize) = 64];
}


message Sheet_RankingDate
{
	repeated E_RankingDate E_RankingDate_List = 1[(yd_fieldoptions.field_arysize)=32];
}

message E_RankingButton
{
	optional int32 m_buttonid = 1[(yd_fieldoptions.field_cname) = "数据ID"];
	optional string m_buttonname = 2[(yd_fieldoptions.field_cname) = "数据名称", (yd_fieldoptions.field_bufsize) = 64];
	optional int32 m_display = 3[(yd_fieldoptions.field_cname) = "是否显示"];
}


message Sheet_RankingButton
{
	repeated E_RankingButton E_RankingButton_List = 1[(yd_fieldoptions.field_arysize)=8];
}
